src/IR.h

```diff
@@ -778,6 +778,11 @@ struct Shuffle : public ExprNode<Shuffle> {
     }
     ///@}
 
+    /** Check if this shuffle is a broadcast of the vector arguments.
+     * returns 0 if the shuffle is not a broadcast, otherwise returns
+     * the broadcast factor. */
+    int is_broadcast() const;
```

Comment:
I think it's a bit confusing when function called is_* returns something other than bool (I can't suggest a very good alternative name though).

Reply:
I thought about this too, but I figured it'd be OK because this does actually work as a bool. 0 is false, and anything else is true, and I'm pretty sure it works as part of any larger boolean expression.

Reply:
I'd still think the name `broadcast_factor()` might be better